doc/matcher-DSL.md

```diff
@@ -0,0 +1,221 @@
+# Matcher DSL
+- [Overview](#Overview)
+- [ClassMatcher](#ClassMatcher)
+- [MethodMatcher](#MethodMatcher)
+
+## Overview
+EaseAgent provides a Matcher DSL (Domain Specific Language) to define a `Points` implementation which is used to defined classes and their methods to be enhanced. 
+
+The design of this DSL is borrowed from bytebuddy's class and method matching DSL design. Why not use bytebuddy's DSL interface directly?
+- Decopule from bytebuddy, which is an excellent bytecode editing framework, but we still want to decouple the API interface from it.
+- Provide a cleaner and easier to use API for Agent scenarios.
+
+
```

Comment:
Need an overview to descript how many matchers we have and what's the purpose of matchers. eg:

We have defined two matchers, there are `ClassMatcher` and `MethodMatcher`.

- The `ClassMatcher` is dedicated to ....
- The `MethodMatcher` is dedicated to ...